ts/packages/anchor/src/coder/borsh/accounts.ts

```diff
@@ -49,7 +49,12 @@ export class BorshAccountsCoder<A extends string = string>
     return Buffer.concat([discriminator, accountData]);
   }
 
-  public decode<T = any>(accountName: A, data: Buffer): T {
+  public decode<T = any>(accountName: A, data: Buffer, skipDiscriminator?: boolean): T {
+    if (skipDiscriminator) {
+      const prepend = Buffer.from("0".repeat(16), "hex");
```

Comment:
Why do we prepend data to accounts before decoding?

Reply:
The `decodeUnchecked` method assumes that the data includes the discriminator. I have changed the logic now, let me know what you think.

Reply:
Thanks, it's clearer now.